src/main/java/gregtech/api/capability/INotifiableHandler.java

```diff
@@ -0,0 +1,46 @@
+package gregtech.api.capability;
+
+import gregtech.api.metatileentity.MetaTileEntity;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+/**
+ * For Item and Fluid handlers capable of notifying entities when
+ * their contents change
+ */
+public interface INotifiableHandler {
```

Comment:
On one hand I don't want to restrict someone's usage of this by constraining it only to one MetaTileEntity to notify but on other hand it does not makes sense to me to hold HashSet in current implementation. Also there is problem that if someone incorrectly overrides `getNotifiableMetaTileEntities` as it generates new hashset every time as I suggested in comment bellow it will prevent correct function of `addNotifiableMetaTileEntity`... This needs to be addressed too